### ISSUE TITLE
`isUserNotificationsOptedIn` - Possible discrepancy between iOS and Android

### ISSUE THREAD

**Issue by ds8k** [urbanairship/react-native-module#349]
### What Airship dependencies are you using?
`"urbanairship-react-native": "11.0.1",`

# Report

### What unexpected behavior are you seeing?
On iOS there is a distinction between `isUserNotificationsEnabled` and `isUserNotificationsOptedIn`, however in the Android SDK `userNotificationsEnabled` *must* be true in order to get the value of system-level permissions

### What is the expected behavior?
Consistency across both platforms - the two values should be distinct. One should be for user notifications, one should be system level opt-in status.

### Relevant code

https://github.com/urbanairship/android-library/blob/27a61106dc543af8a2f453246179958d99feebf5/urbanairship-core/src/main/java/com/urbanairship/push/PushManager.java#L696-L703

```java
  /**
   * Checks if notifications are enabled for the app and in the push manager.
   *
   * @return {@code true} if notifications are opted in, otherwise {@code false}.
   */
  public boolean areNotificationsOptedIn() {
      return getUserNotificationsEnabled() && notificationManagerCompat.areNotificationsEnabled();
  }
```

**Comment by rlepinski** [comment]
@ds8k Work will start shortly:

Going to add:
- `isUserNotificationsEnabledInSystemSettings`  for system level check. Naming on this one is hard so if you have a better name let me here it :)
- Android method `getNotificationChannelStatus(channel)` that returns `enabled`, `disabled`, `unavailable` where `unavailable` is returned if the notification is not yet created.

**Comment by rlepinski** [comment]
This was resolved in 12.1.0
